### PR TITLE
fixes #13828 - CVE-2016-2100 - only show bookmarks in administer if user can edit bookmarks

### PR DESCRIPTION
**Pull request by unorthodoxgeek**

### CODE REVIEW

File: app/services/menu/loader.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def self.load
             menu.item :roles,            :caption => N_('Roles')
           end
           menu.divider
-          menu.item :bookmarks,          :caption => N_('Bookmarks')
+          menu.item :bookmarks,          :caption => N_('Bookmarks') if User.current && User.current.can?(:edit_bookmarks)
```
Comment (ohadlevy) [inline app/services/menu/loader.rb:37]: what about the controller? shouldn't it use the my_bookmarks scope vs all?

Comment (unorthodoxgeek) [inline app/services/menu/loader.rb:37]: question is - what if the user is an admin? shouldn't they see everything? also, "my bookmarks" are very badly defined, there's no way to define it using taxonomies, etc...
I think we should solve the CVE, and think about redesigning bookmarks' permissions

**Comment by domcleal** [comment]
The CVE identifier was changed to CVE-2016-2100, the commit message should be changed accordingly on your next update.

File: app/services/menu/loader.rb (hunk 1 above)
Comment (domcleal) [inline app/services/menu/loader.rb:37]: a) This is completely in the wrong place, the loader shouldn't be checking permissions - it has no user context.  Permissions are checked per user using the routes/access_permissions list.
b) Changing the menu does nothing to fix the security issues in the ticket, the routes are still fully accessible and vulnerable.
c) The my_bookmarks scope serves its purpose to limit visibility of resources, but it's not used in the vulnerable controllers.  Adding taxonomies isn't really relevant to fixing this.

**Comment by unorthodoxgeek** [comment]
closing as my solution is wrong. don't know how to solve it.

**Pull request closed without merging**